### PR TITLE
Unpin some dependencies

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Unpin some dependencies in package.json

#### Testing instructions

* Verify there are no packages removed/added in `yarn.lock`
* Validate the dependency tree has not changed:
    1. Install dependencies and generate the tree for this branch: `find . -type d -name node_modules | xargs rm -fr && yarn && NODE_OPTIONS=--max_old_space_size=8192 yarn run effective-module-tree -o list > ../branch-tree`
    2. Install dependencies and generate the tree for master (tested with c9aab3df58): `find . -type d -name node_modules | xargs rm -fr && yarn && NODE_OPTIONS=--max_old_space_size=8192 yarn run effective-module-tree -o list > ../master-tree`
    3. Diff both trees and check there are no differences: `diff ../branch-tree ../master-tree`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -235,126 +235,123 @@
...
+		"fsevents": "^2.1.2"
 	},
 	"husky": {
 		"hooks": {
 			"pre-commit": "yarn run -s install-if-no-packages && node bin/pre-commit-hook.js",
 			"pre-push": "yarn run -s install-if-no-packages && node bin/pre-push-hook.js"
 		}
-	},
-	"devDependencies": {
-		"check-npm-client": "1.1.1"
```
Comment (scinos) [inline package.json:358]: Deleted because it is already a regular dependency

**Comment by sgomes** [comment]
This produces a large number of differences for me, I'm afraid. Some of the most significant ones relate to Babel, but are somewhat surprising, because they don't seem to match what I remembered from the `master` branch (I used the same commit as you):

master:
`wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @babel/core@7.8.4 @babel/code-frame@7.8.3 @babel/highlight@7.8.3`

branch:
`wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @babel/core@7.9.0 @babel/code-frame@7.8.3 @babel/highlight@7.9.0`

I followed your exact sequence of steps in generating the files for the diffs, so I'm not sure why I got different results.

**Comment by sgomes** [comment]
Looks like the base commit is no longer the one in the PR description, but rather db4f010197b3f42426e16d0b4416a91d008c5aa8. I'll try again using that one.

**Comment by sgomes** [comment]
Alright, the new base commit produces a much better output! 🙂 There's still a significant number of differences, but they all seem related to `mkdirp` going from `0.5.1` to `0.5.5`. E.g.

```diff
3606,3607c3606,3607
< wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @wordpress/dependency-extraction-webpack-plugin@2.4.0 webpack@4.42.0 mkdirp@0.5.1
< wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @wordpress/dependency-extraction-webpack-plugin@2.4.0 webpack@4.42.0 mkdirp@0.5.1 minimist@0.0.8
---
> wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @wordpress/dependency-extraction-webpack-plugin@2.4.0 webpack@4.42.0 mkdirp@0.5.5
> wp-calypso-monorepo@0.17.0 @automattic/calypso-build@6.1.0 @wordpress/dependency-extraction-webpack-plugin@2.4.0 webpack@4.42.0 mkdirp@0.5.5 minimist@1.2.5
```

That seems safe enough to me 👍

**Review by sgomes — APPROVED** [review]
LGTM! The only differences I see are related to a revision update of `mkdirp`, which should be safe enough, considering it's:
a) a revision update
b) a build-time tool

**Comment by scinos** [comment]
@sgomes Thanks for the review, and sorry for the confusion :( As you said, I rebased this branch but got distracted when something else while updating the instructions in this PR. The commit to check against should be `db4f010197`.

That being said, this should have exactly 0 differences. I'll investigate what's going on with `mkdirp` before merging (probably a bad rebase)

**Comment by scinos** [comment]
Alright, branch is cleaner now.

There is still one (and only one) diff in the trees:
```diff
2964027,2964028c2964027,2964028
< wp-calypso-monorepo@0.17.0 mkdirp@0.5.1
< wp-calypso-monorepo@0.17.0 mkdirp@0.5.1 minimist@0.0.8
---
> wp-calypso-monorepo@0.17.0 mkdirp@0.5.5
> wp-calypso-monorepo@0.17.0 mkdirp@0.5.5 minimist@1.2.5
```

This is unavoidable. When unpinning `mkdirp@0.5.1` to `mkdirp@^0.5.1`, turns out that `mkdirp@^0.5.1` was already used in our dep tree and resolved with `mkdirp@0.5.5`. So there is no way we can depend on `mkdirp@^0.5.1` on our package.json and get `mkdirp@0.5.1` without downgrading lots of dependencies from `0.5.5` to `0.5.1` (which is what @sgomes found in the previous comment)

**Comment by sgomes** [comment]
> (which is what @sgomes found in the previous comment)

You're right, looks like I diff'ed in the wrong order 😅 What I found was a downgrade, not an upgrade!

**Pull request merged**